pandas/tests/frame/test_arithmetic.py

```diff
@@ -211,6 +213,17 @@ def test_df_sub_datetime64_not_ns(self):
                                  pd.Timedelta(days=2)])
         tm.assert_frame_equal(res, expected)
 
+    @pytest.mark.parametrize('other', [
+        pd.Timedelta('1d'),
+        datetime.timedelta(days=1),
+        np.timedelta64(1, 'D')
+    ])
+    def test_timestamp_df_add_timedelta(self, other):
+        # GH 22005
+        expected = pd.DataFrame([pd.Timestamp('2018-01-02')])
```

Comment:
can you add another test that uses a dataframe of mixed types (e.g. just add a string column and an integer column)

Reply:
int or str can't operate with Timedelta. Even if they are mixed in one DataFrame, they use different Block, won't be affect by DatetimeBlock here.
> `pd.Timedelta('1d') + pd.DataFrame([pd.Timestamp('2018-01-01'), 123])`
TypeError: Could not operate Timedelta('1 days 00:00:00') with block values unsupported operand type(s) for +: 'Timedelta' and 'int'

Reply:
> int or str can't operate with Timedelta. 

The idea is 1) for the mixed-dtype cases that are valid, test them, 2) for (a few of) the mixed-dtype cases that are not, test that they raise as expected.